srs improvements

mhines-usgs opened the pull request:
moving two general functions from formatSRS into their own functions

mhines-usgs committed:
srs improvements

mhines-usgs committed:
changing reportObject to reflect that its only recieving the readings

mhines-usgs committed:
noticed i forgot to add this to the test file

mhines-usgs committed:
Merge remote-tracking branch 'upstream/master'

mhines-usgs committed:
Merge remote-tracking branch 'upstream/master'

mhines-usgs committed:
updates for srs to use more general functions kathy made. still working out the Rmd qualifiers summary table

mhines-usgs committed:
only send in the three columns needed for the formatting of the qualifiers

mhines-usgs commented:
This isn't ready to merge yet, but getting close. Need to address problems in the summary qualifiers table showing up after using the generic function and also need to redo existing test to use generic functions

mhines-usgs committed:
merge conflict

mhines-usgs committed:
fix tests and logic

mhines-usgs committed:
forgot data file, small docu change and removed white space

mhines-usgs committed:
found an issue in the comment section (was comparing time, not date) and added a test

mhines-usgs committed:
tests tests tests

mhines-usgs committed:
tested tests and one was incorrect, fixing.

mhines-usgs committed:
merge conflict

Merge branch 'master' of github.com:mhines-usgs/repgen

# Conflicts:
#	tests/testthat/test-sensorreadingsummary.R

mhines-usgs committed:
test for getAppliedCorrection

mhines-usgs committed:
last srs test

zmoore-usgs commented on inst/sensorreading/sensorreading.Rmd:
```diff
@@ -63,10 +63,14 @@ if(!identical(tbl, "The dataset requested is empty.")){
+    tbl2 <- formatQualifiersTable(quals)
     if(!is.null(tbl2) && nrow(tbl2) > 0){
-      kable(tbl2, align='l', row.names=FALSE, caption = "Qualifiers")
+      if(!is.null(tbl2) && nrow(tbl2) > 0){
```
Is this a duplicate of the line above?

mhines-usgs replied:
I guess so, I left it here from before, but I can try removing it and see if it matters or not. I find it hard to work with the Rmd files because I haven't figured out how really to debug within them

mhines-usgs replied:
Actually this might have been introduced by me from a merge conflict, oops. Fixed.

zmoore-usgs commented on R/sensorreading-data.R:
```diff
@@ -264,17 +235,17 @@ getAppliedCorrection <- function(raw, corrected) {
 #' 
-getCorrectedRef <- function (value, nearestCorrectedValue, uncertainty) {
-  if ((!isEmpty(value)) && (!isEmpty(uncertainty)) && (!isEmpty(nearestCorrectedValue))) {
+getCorrectedRef <- function (value, nearestcorrectedValue, uncertainty) {
```
Just a little styling question: Should nearestcorrectedValue be camel cased to nearestCorrectedValue?

mhines-usgs replied:
I actually had it that way earlier, and then I realized in the JSON its nearestcorrectedValue

but I could be swayed? :)

zmoore-usgs replied:
Personally I'd say that the JSON should be corrected to be properly camel cased and that the variable should be the same way. But it doesn't really matter all that much. @thongsav-usgs ?

mhines-usgs replied:
I'll make a future task for this, I don't think it's urgent and it would be good to do them together when we don't have a hotfix floating around? @thongsav-usgs

mhines-usgs replied:
https://internal.cida.usgs.gov/jira/browse/AQCU-1015 for the JSON change and changes to sensorreading-data.R file

zmoore-usgs commented on R/utils-html.R:
```diff
@@ -193,7 +193,7 @@ getComments <- function(comments) {
 #' @return list of deduplicated qualifiers with column names.
 formatQualifiersTable <- function(inQualifiers) {
-    if (isEmptyOrBlank(inQualifiers) || nrow(inQualifiers) == 0) return()
+    if(isEmptyOrBlank(inQualifiers) || (isEmptyVar(inQualifiers))) return()
```
One thing that Lindsay suggested on my stuff was to limit functions to only having a single return. I guess maybe that should be the case here? Maybe wrap the rest of the function in a negated version of this if statement?

mhines-usgs replied:
yes, I see that this is pretty close to being what we need how you like so I'll get it to work with one return. Thanks!

zmoore-usgs commented on R/utils-html.R:
```diff
@@ -210,10 +210,10 @@ formatQualifiersTable <- function(inQualifiers) {
 #' @return comma-delimited string of qualifier codes
 formatQualifiersStringList <- function(inQualifiers) {
-  if(length(inQualifiers[[1]]) < 1) return("");
+  if(isEmptyOrBlank(inQualifiers) || (isEmptyVar(inQualifiers))) return("");
```
Same as above.

zmoore-usgs reviewed (commented):
Overall changes look good to me! Just a couple of small comments

mhines-usgs committed:
removing redundant line from a badly resolved merge conflict

mhines-usgs committed:
updates from Zack's peer review
modified returns so only one existed for each function, and therefore had to modify an existing test for qualifiers that was checking for null, had to change because dataframes don't return true for null because they're not null. checking instead if there are no rows in the dataframe, because it is created and returned without rows if there are no qualifiers now.

mhines-usgs commented:
re: dataframes not returning true for is.null when they have no rows: http://stackoverflow.com/questions/28556658/why-does-an-empty-dataframe-fail-an-is-null-test

mhines-usgs commented on R/utils-html.R:
```diff
@@ -193,15 +193,16 @@ getComments <- function(comments) {
-        columnNames <- c("Code",
-                         "Identifier",
-                         "Description")
+  toRet <- data.frame()
```
create an empty data frame so we can return empty if the if block is skipped

mhines-usgs commented on inst/sitevisitpeak/sitevisitpeak.Rmd:
```diff
@@ -71,10 +71,8 @@ if(!identical(tbl, "The dataset requested is empty.")){
-    if(!is.null(tbl2) && nrow(tbl2) > 0){
-      kable(tbl2, align='l', row.names=FALSE, caption = "Qualifiers")
-    }
+  if(!isEmptyOrBlank(tbl2)){
```
suffering from the same redundant if block here so modified it

lindsayplatt commented:
@mhines-usgs that's why isEmptyOrBlank also checks nrow

mhines-usgs commented on tests/testthat/test-utils-html.R:
```diff
@@ -135,7 +135,7 @@ test_that('formatQualifiersTable handles no qualifiers', {
   qualsDf <- repgen:::readQualifiers(inQualifiers,NULL)
   qualList <- repgen:::formatQualifiersTable(qualsDf)
-  expect_true(is.null(qualList))
+  expect_true(nrow(qualList)==0)
```
had to change this test to check for a dataframe with 0 rows instead of null since dataframes can't be tested for null http://stackoverflow.com/questions/28556658/why-does-an-empty-dataframe-fail-an-is-null-test
